--- a/frontend/node-env.nix	2022-09-15 12:06:24.370895495 +0800
+++ b/frontend/node-env.nix	2022-09-15 12:08:59.411040273 +0800
@@ -372,6 +372,9 @@
             node ${reconstructPackageLock}
           ''}
 
+          export NG_CLI_ANALYTICS=ci
+          export DEBUG='cypress:*'
+          export CYPRESS_INSTALL_BINARY=0
           node ${addIntegrityFieldsScript}
         ''}
 
@@ -520,6 +523,11 @@
 
           source $includeScriptPath
 
+          chmod a+x ./node_modules/node-gyp-build/bin.js
+          chmod a+x ./node_modules/node-gyp-build/optional.js
+          chmod a+x ./node_modules/node-gyp-build/build-test.js
+          patchShebangs ./node_modules/node-gyp-build/
+
           # Create fake package.json to make the npm commands work properly
           cp ${src}/package.json .
           chmod 644 package.json
@@ -543,6 +551,14 @@
 
           mv ${packageName} lib
           ln -s $out/lib/node_modules/.bin $out/bin
+          chmod a+x $out/lib/node_modules/.bin/*
+          for FILE in $(find $out/lib/node_modules/.bin/ -exec readlink -f {} \;); do
+            chmod a+x $FILE
+            patchShebangs $FILE
+            if [ "$(basename $FILE)" == "ng.js" ]; then
+              ls -la $FILE
+            fi
+          done
         '';
       } // extraArgs);
 
